packages/runtime/test-runtime-utils/api-report/test-runtime-utils.legacy.alpha.api.md

```diff
@@ -301,8 +301,6 @@ export class MockFluidDataStoreContext implements IFluidDataStoreContext {
     // (undocumented)
     deltaManager: IDeltaManager<ISequencedDocumentMessage, IDocumentMessage>;
     // (undocumented)
-    ensureNoDataModelChanges<T>(callback: () => T): T;
```

Comment:
This technically should have been marked as deprecated before now. To be completely strict we would not remove this from the mock until properly deprecated. This is test-runtime-utils and maybe there is a different contract there. I will let API CR make final call.

Reply:
Unless there is a reason we _have_ to remove this in this PR _and_ we are confident that _none_ of our partners are using it (beyond just Loop), then I think it would be best to stick to the process and retain this as deprecated for future removal.

Reply:
So, it seems like "ensureNoDataModelChanges" was deprecated on the interface and now he is removing it from the interface and classes. 
Is it wrong to do so? In order to remove it, should it be marked as deprecated on interfaces and all classes implementing that interface?

Reply:
@jason-ha and @Josmithr Please take a look at this comment when you get time. We need to complete this breaking change PR by next week. Thanks.

Reply:
Sorry, missed the earlier responses. Yes, strictly speaking any implementation of the interface that declares the same properties should also mark the members as deprecated. There is nothing stopping `MockFluidDataStoreContext` from continuing to expose a `ensureNoDataModelChanges` member even if the interface it extends plans to remove it from the interface. So, to be explicit, any implementations for which the member is deprecated should be marked so as well.

Unless there is an urgent need to remove this, it would probably be better to mark the appropriate implementation members as `@deprecated` and add them to the issue tracking `@legacy` breaks we plan to make in `2.20`.

I don't think there is anything stopping us from removing the deprecated interface member, though.

Reply:
I would be okay making an exception for this one since the expected use is for mock to be used in places that would use `ensureNoDataModelChanges` via interface. Technically, we should have deprecated before and should wait, but I would be surprised if there is fallout.

Reply:
> @Josmithr, are you also okay with this exception? Latest version has this removed per @jatgarg's suggestion to go the exception route.

I don't have a strong opinion either way. Introducing a break to something with "Mock" in the name seems safer than most alternatives, so I don't see any harm in it. But I do think we should generally try to be consistent with this stuff. 🤷‍♂️